receiver/mongodbreceiver/metrics.go

```diff
@@ -44,142 +44,152 @@ var documentMap = map[string]metadata.AttributeOperation{
 	storageSizePath := []string{"storageSize"}
 	storageSize, err := dig(doc, storageSizePath)
 	if err != nil {
-		errors.AddPartial(1, err)
+		errs.AddPartial(1, err)
 		return
 	}
 	storageSizeValue, err := parseInt(storageSize)
 	if err != nil {
-		errors.AddPartial(1, err)
+		errs.AddPartial(1, err)
 		return
 	}
 	s.mb.RecordMongodbStorageSizeDataPoint(now, storageSizeValue, dbName)
 }
 
-func (s *mongodbScraper) recordObjectCount(now pcommon.Timestamp, doc bson.M, dbName string, errors scrapererror.ScrapeErrors) {
+func (s *mongodbScraper) recordObjectCount(now pcommon.Timestamp, doc bson.M, dbName string, errs *scrapererror.ScrapeErrors) {
 	objectsPath := []string{"objects"}
 	objects, err := dig(doc, objectsPath)
 	if err != nil {
-		errors.AddPartial(1, err)
+		errs.AddPartial(1, err)
 		return
 	}
 	objectsVal, err := parseInt(objects)
 	if err != nil {
-		errors.AddPartial(1, err)
+		errs.AddPartial(1, err)
 		return
 	}
 	s.mb.RecordMongodbObjectCountDataPoint(now, objectsVal, dbName)
 }
 
-func (s *mongodbScraper) recordIndexCount(now pcommon.Timestamp, doc bson.M, dbName string, errors scrapererror.ScrapeErrors) {
+func (s *mongodbScraper) recordIndexCount(now pcommon.Timestamp, doc bson.M, dbName string, errs *scrapererror.ScrapeErrors) {
 	indexesPath := []string{"indexes"}
 	indexes, err := dig(doc, indexesPath)
 	if err != nil {
-		errors.AddPartial(1, err)
+		errs.AddPartial(1, err)
 		return
 	}
 	indexesVal, err := parseInt(indexes)
 	if err != nil {
-		errors.AddPartial(1, err)
+		errs.AddPartial(1, err)
 		return
 	}
 	s.mb.RecordMongodbIndexCountDataPoint(now, indexesVal, dbName)
 }
 
-func (s *mongodbScraper) recordIndexSize(now pcommon.Timestamp, doc bson.M, dbName string, errors scrapererror.ScrapeErrors) {
+func (s *mongodbScraper) recordIndexSize(now pcommon.Timestamp, doc bson.M, dbName string, errs *scrapererror.ScrapeErrors) {
 	indexSizePath := []string{"indexSize"}
 	indexSize, err := dig(doc, indexSizePath)
 	if err != nil {
-		errors.AddPartial(1, err)
+		errs.AddPartial(1, err)
 		return
 	}
 	indexSizeVal, err := parseInt(indexSize)
 	if err != nil {
-		errors.AddPartial(1, err)
+		errs.AddPartial(1, err)
 		return
 	}
 	s.mb.RecordMongodbIndexSizeDataPoint(now, indexSizeVal, dbName)
 }
 
-func (s *mongodbScraper) recordExtentCount(now pcommon.Timestamp, doc bson.M, dbName string, errors scrapererror.ScrapeErrors) {
-	extentsPath := []string{"numExtents"}
-	extents, err := dig(doc, extentsPath)
-	if err != nil {
-		errors.AddPartial(1, err)
-		return
-	}
-	extentsVal, err := parseInt(extents)
-	if err != nil {
-		errors.AddPartial(1, err)
-		return
+func (s *mongodbScraper) recordExtentCount(now pcommon.Timestamp, doc bson.M, dbName string, errs *scrapererror.ScrapeErrors) {
+	// Mongo version 4.4+ no longer returns numExtents since it is part of the obsolete MMAPv1
+	// https://www.mongodb.com/docs/manual/release-notes/4.4-compatibility/#mmapv1-cleanup
+	mongo44, _ := version.NewVersion("4.4")
```

Comment:
Please submit these version-related changes in a separate PR.

Reply:
will do